src/TConsole.cpp

```diff
@@ -2949,3 +2950,41 @@ void TConsole::setProfileName(const QString& newName)
         pC->setProfileName(newName);
     }
 }
+
+
+void TConsole::dragEnterEvent(QDragEnterEvent* e)
+{
+    if (e->mimeData()->hasUrls()) {
+        e->acceptProposedAction();
+    }
+}
+
+//https://amin-ahmadi.com/2016/01/04/qt-drag-drop-files-images/
+void TConsole::dropEvent(QDropEvent* e)
+{
+    QStringList accepted_types;
+    accepted_types << "jpeg"
+                   << "jpg"
+                   << "png";
+    foreach(const QUrl& url, e->mimeData()->urls()) {
```

Comment:
```suggestion
    for (const auto& url : e->mimeData()->urls()) {
```

`foreach` is and Qt-specific, predates the `for` which is now in the standard